### PR TITLE
feat: added support for auto type in dev mode

### PR DESCRIPTION
**Pull request by a3mitskevich**

### CODE REVIEW

File: src/extractor.ts [hunk 1]
```diff
@@ -39,51 +42,65 @@ import { createInternalLogger } from './internal-loger'
...
-    withWhitespace: true,
-    ligatures: [],
-  }
+  const fontServeProxy = new Map<string, () => Promise<ServeFontStubResponse | null>>()
+
+  const glyphsFindMap = new Map<string, string[]>()
+
+  const autoTarget = new Proxy<Required<Target>>(
+    {
+      fontName: '[auto font target have not current font name]',
```
Comment (PlisMikhail) [inline src/extractor.ts:63]: maybe you need more informative message?

File: src/extractor.ts [hunk 2]
```diff
@@ -376,7 +459,7 @@ export default function FontExtractor (pluginOption: PluginOption = { type: 'aut
           const options = optionsMap.get(name)
 
           if (!options) {
-            logger.warn(`Font "${name}" has no minify options`)
+            logger.warn(`Font "${name}" hasn't minify options`)
```
Comment (PlisMikhail) [inline src/extractor.ts:462]: image

File: src/extractor.ts [hunk 3]
```diff
@@ -212,29 +292,21 @@ export default function FontExtractor (pluginOption: PluginOption = { type: 'aut
...
-      if (mode === 'auto') {
-        logger.warn('Serve minification disabled in "auto" mode')
-        return code
-      }
-
-      const minifiedBuffers = await processMinify(
-        font.name,
-        font.aliases.map<MinifyFontOptions>(url => ({
+      font.aliases.forEach(url => {
+        if (fontServeProxy.has(url)) {
```
[the remark this answers is not in the document]
Comment (a3mitskevich) [reply src/extractor.ts:302]: fixed but for memo fn for plain font

File: src/utils.ts [hunk 4]
```diff
@@ -27,6 +29,15 @@ export function intersection<T> (array1: T[], array2: T[]): T[] {
   return array1.filter(item => array2.includes(item))
 }
 
+export function hasDifferent<T> (array1: T[], array2: T[]): boolean {
+  if (array1.length !== array2.length) {
+    return true
+  }
+  const biggest = array1.length > array2.length ? array1 : array2
```
[the remark this answers is not in the document]
Comment (a3mitskevich) [reply src/utils.ts]: changed

File: src/extractor.ts (hunk 2 above)
Comment (a3mitskevich) [reply src/extractor.ts]: fixed

File: src/extractor.ts (hunk 1 above)
Comment (a3mitskevich) [reply src/extractor.ts]: Changed

**Review by PlisMikhail — APPROVED** [review]

**Pull request merged**